src/idl_gen_js.cpp

```diff
@@ -110,16 +144,23 @@ class JsGenerator : public BaseGenerator {
   std::string &exports = *exports_ptr;
   for (auto it = sorted_namespaces.begin();
        it != sorted_namespaces.end(); it++) {
-    code += "/**\n * @const\n * @namespace\n */\n";
-    if (it->find('.') == std::string::npos) {
-      code += "var ";
-      if(parser_.opts.use_goog_js_export_format) {
-        exports += "goog.exportSymbol('" + *it + "', " + *it + ");\n";
-      } else {
-        exports += "this." + *it + " = " + *it + ";\n";
+    if (ts_) {
+      if (it->find('.') == std::string::npos) {
+        code += "import { flatbuffers } from \"./flatbuffers\"\n";
```

Comment:
the original PR also has this import emit in a loop, but I wasn't able to figure out why: it seems to be that it only needs to be emitted once?

also, to make the code self-documenting, it might be worth it to assign `it->find('.') == std::string::npos` to a local bool variable that indicates what that condition is in fact testing